tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/ConfusionMatrixConditionCount.java

```diff
@@ -0,0 +1,187 @@
+  }
+  /**
+   * Creates a ConfusionMatrixConditionCount type of Metric
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of the metric, if null then {@link Class#getSimpleName()} is used
+   * @param confusionMatrixCond the confusion matrix condition to calculate
+   * @param threshold a threshold value in <code>[0, 1]</code>. A threshold is compared with
+   *     prediction values to determine the truth value of predictions (i.e., above the threshold is
+   *     <code>true</code>, below is <code>false</code>). One metric value is generated for each
+   *     threshold value.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the data type for the variables
+   */
+  public ConfusionMatrixConditionCount(
+      Ops tf,
+      String name,
+      ConfusionMatrixEnum confusionMatrixCond,
+      float threshold,
+      long seed,
+      Class<T> type) {
+    this(tf, name, confusionMatrixCond, new float[] {threshold}, seed, type);
+  }
+
+  /**
+   * Creates a ConfusionMatrixConditionCount type of Metric
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of the metric, if null then {@link Class#getSimpleName()} is used
+   * @param confusionMatrixCond the confusion matrix condition to calculate
+   * @param thresholds threshold values in <code>[0, 1]</code>. A threshold is compared with
+   *     prediction values to determine the truth value of predictions (i.e., above the threshold is
+   *     <code>true</code>, below is <code>false</code>). One metric value is generated for each
+   *     threshold value.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the data type for the variables
+   */
+  public ConfusionMatrixConditionCount(
+      Ops tf,
+      String name,
+      ConfusionMatrixEnum confusionMatrixCond,
+      float[] thresholds,
+      long seed,
+      Class<T> type) {
+    super(tf, name, seed);
+    accumulatorName = this.getVariableName(ACCUMULATOR);
+    this.type = type;
+    this.confusionMatrixCond = confusionMatrixCond;
+    this.thresholds = thresholds;
+    init();
+  }
+
+  private void init() {
+    Shape variableShape = Shape.of(this.thresholds.length);
+
+    Zeros<T> zeros = new Zeros<>(getTF());
+    accumulator =
+        getTF()
+            .withName(getAccumulatorName())
+            .variable(zeros.call(getTF().constant(variableShape), type));
+    initializer = getTF().assign(accumulator, zeros.call(getTF().constant(variableShape), type));
+  }
+
+  /**
+   * Gets the initializer for the accumulator variable
+   *
+   * @return the initializer for the accumulator variable
+   */
+  public Assign<T> getInitializer() {
+    return initializer;
+  }
+
+  /** {@inheritDoc} */
```

Comment:
The underlying `MetricsHelper.updateConfusionMatrixVariables` supports and realigns complex dimensional relationships between `labels` and `predictions`. Seems like we should document what is permitted and how it is interpreted?